fairscale/experimental/wgit/signal_sparsity.py

```diff
@@ -63,17 +82,41 @@ def _validate_conf(self) -> None:
 
         this should assert fail if checking fails.
         """
-        pass
+        # assert that both top_k_elements and top_k_percent aren't set for sst and dst
+        assert (
+            self._sst_top_k_element is None or self._sst_top_k_percent is None
+        ), "Both top-k element and top-k percent has been provided as argument"
+        assert (
+            self._dst_top_k_element is None or self._dst_top_k_percent is None
+        ), "Both top-k element and top-k percent has been provided as argument"
 
     def dense_to_sst(self, dense: Tensor) -> Tensor:
         """Get SST from a tensor
 
         Dense -> fft -> top-k -> results.
 
+        For sparsification (or getting the topk) of the tensor in the transformed domain, the sparse
+        mask needs to be generated from the absolute values of the transformed tensors. However, using
+        the mask, we mask out the top_k values from the transformed tensor `w_fft` itself instead.
+        Note: the transformed tensor `w_fft` is a tensor of complex values, whereas `w_fft_abs` is a
+        tensor with real values from which we generate the sparse mask.
+
         Returns:
             Same shaped tensor, still in dense format but in frequency domain and has zeros.
         """
-        pass
+        if self._sst_top_k_element is not None:
+            assert self._sst_top_k_dim is None or (
```

Comment:
this check can be in the validate_conf? this doesn't depend on the actually size of the dimension of the tensor, thus, it can be checked earlier?

Reply:
I presume this check cannot be moved into the validate_conf, since we need the input tensor dim. However, it can be moved to the `_sparse_mask` method as you suggest in the later comment. I have moved it there for the next commit.

Reply:
What I meant here is the check of "is None", not the next line after the "or" part. The none check does not need the input tensor dim, right?

Reply:
The none check is used together with the rest of the assetion, which checks `_sst_topk_element <= in_tensor.shape[self._sst_top_k_dim]`. If `self._sst_top_k_dim`, then that would be an error for the tensor.shape method. We are either checking if we have one of dim=None, or the rest of the assert.

Reply:
Right, I miss read it. I thought you are check top_k_percent not None here. Sorry.

Where do you check one and only one of (element, percent) is not None? I see validate_conf only checking they not both not None, right?